drivers/adc/ad469x/ad469x.c

```diff
@@ -0,0 +1,431 @@
+					     AD469x_SEQ_CTRL_NUM_SLOTS_AS_MASK,
+					     AD469x_SEQ_CTRL_NUM_SLOTS_AS(1));
+		break;
+
+	default:
+		return FAILURE;
+		break;
+	}
+
+	return ret;
+}
+
+/**
+ * @brief Configure converter busy indicator to the output of the specified port
+ * @param [in] dev - ad469x_dev device handler.
+ * @param [in] gp_sel - Port.
+ * @return \ref SUCCESS in case of success, \ref FAILURE otherwise.
+ */
+int32_t ad469x_set_busy(struct ad469x_dev *dev,
+			enum ad469x_busy_gp_sel gp_sel)
+{
+	int32_t ret;
+
+	ret = ad469x_spi_write_mask(dev,
+				    AD469x_REG_GP_MODE,
+				    AD469x_GP_MODE_BUSY_GP_EN_MASK,
+				    AD469x_GP_MODE_BUSY_GP_EN(1));
+
+	ret |= ad469x_spi_write_mask(dev,
+				     AD469x_REG_GP_MODE,
+				     AD469x_GP_MODE_BUSY_GP_SEL_MASK,
+				     AD469x_GP_MODE_BUSY_GP_SEL(gp_sel));
+
+	return ret;
+}
+
+/**
+ * Initialize the device.
+ * @param [out] device - The device structure.
+ * @param [in] init_param - The structure that contains the device initial
+ * 		parameters.
+ * @return \ref SUCCESS in case of success, \ref FAILURE otherwise.
+ */
+int32_t ad469x_init(struct ad469x_dev **device,
+		    struct ad469x_init_param *init_param)
+{
+	struct ad469x_dev *dev;
+	int32_t ret;
+	uint8_t data = 0;
+	struct spi_engine_init_param *spi_eng_init_param =
+			init_param->spi_init.extra;
+
+	dev = (struct ad469x_dev *)malloc(sizeof(*dev));
+	if (!dev)
+		return FAILURE;
+
+	ret = axi_clkgen_init(&dev->clkgen, &init_param->clkgen_init);
+	if (ret != SUCCESS) {
+		printf("error: %s: axi_clkgen_init() failed\n",
+		       init_param->clkgen_init.name);
+		goto error_dev;
+	}
+
+	ret = axi_clkgen_set_rate(dev->clkgen, init_param->axi_clkgen_rate);
+	if (ret != SUCCESS) {
+		printf("error: %s: axi_clkgen_set_rate() failed\n",
+		       init_param->clkgen_init.name);
+		goto error_clkgen;
+	}
+
+	ret = ad469x_init_gpio(dev, init_param);
+	if (ret != SUCCESS)
+		goto error_clkgen;
+
+	ret = spi_init(&dev->spi_desc, &init_param->spi_init);
+	if (ret != SUCCESS)
+		goto error_gpio;
+
+	dev->reg_access_speed = init_param->reg_access_speed;
+	dev->dev_id = init_param->dev_id;
+
+	/* 8 biti data width for register access */
+	ret = spi_engine_set_transfer_width(dev->spi_desc, 8);
+	if (ret != SUCCESS)
+		goto error_spi;
+
+	spi_engine_set_speed(dev->spi_desc, dev->reg_access_speed);
+
+	data = 0xEA;
+	ret = ad469x_spi_reg_write(dev, AD469x_REG_SCRATCH_PAD, data);
+	if (ret < 0)
+		goto error_spi;
+
+	ad469x_spi_reg_read(dev, AD469x_REG_SCRATCH_PAD, &data);
+	if (ret != SUCCESS)
+		goto error_spi;
+
+	ret = ad469x_set_busy(dev, AD469x_busy_gp0);
+	if (ret != SUCCESS)
+		goto error_spi;
+
+	ret = ad469x_set_channel_sequence(dev, AD469x_single_cycle);
+	ad469x_enter_conversion_mode(dev);
+	if (ret != SUCCESS)
+		goto error_spi;
+
+	ret = spi_engine_set_transfer_width(dev->spi_desc,
+					    spi_eng_init_param->data_width);
+	if (ret != SUCCESS)
+		goto error_spi;
+
+	spi_engine_set_speed(dev->spi_desc, dev->spi_desc->max_speed_hz);
```

Comment:
I see that to write to register the spi_engine_speed is changed.
If in the app there is a call to ad469x_spi_reg_read, spi_speed must be changed too?

Reply:
Yes, solved by changing speed to every register access